site/helm/installing.md

```diff
@@ -0,0 +1,169 @@
+[flux-helm-test](https://github.com/weaveworks/flux-helm-test).
+
+ 1. Fork [flux-helm-test](https://github.com/weaveworks/flux-helm-test)
+    on Github and
+ 1. install Weave Flux and its Helm Operator by specifying your fork
+    URL.
+
+*Just make sure you replace `YOURUSER` with your GitHub username in
+the command below:*
+
+```sh
+helm install --name flux \
+--set helmOperator.create=true \
+--set git.url=git@github.com:YOURUSER/flux-helm-test \
+--set git.chartsPath=charts \
+--namespace flux \
+weaveworks/flux
+```
+
+Allow some time for all containers to get up and running. If you're
+impatient, run the following command and see the pod creation
+process.
+
+```sh
+watch kubectl get pods --all-namespaces
+```
+
+You will notice that `flux` and `flux-helm-operator` will start
+turning up in the `flux` namespace.
+
+## Giving write access
+
+For the real benefits of GitOps, Flux will need acccess to your
+git repository to update configuration if necessary. To facilitate
+that you will need to add a deploy key to your fork of the
+repository.
+
+This is pretty straight-forward as Flux generates a SSH key and
+logs the public key at startup. Find the SSH public key with:
+
+```sh
+export FLUX_POD=$(kubectl get pods --namespace flux -l "app=flux,release=flux" -o jsonpath="{.items[0].metadata.name}")
+kubectl -n flux logs $FLUX_POD | grep identity.pub | cut -d '"' -f2
+```
+
+In order to sync your cluster state with git you need to copy the
+public key and create a deploy key with write access on your GitHub
+repository.
+
+Open GitHub, navigate to your fork, go to **Setting > Deploy keys**(`https://github.com/YOURUSER/flux-example/settings/keys/new`),
+click on **Add deploy key**, check **Allow write access**, paste
+the Flux public key and click **Add key**.
+
+Once Flux has confirmed access to the repository, it will start
+deploying the workloads of `flux-helm-test`. After a while you
+will be able to see the Helm releases listed like so:
+
+```sh
+helm list --namespace test
+```
+
+## Committing a small change
+
+`flux-helm-test` is a very simple example in which two services
+(mongodb and mariadb) are deployed. Here we will simply update the
+version of mongodb to a newer version to see if Flux will pick this
+up and update our cluster.
+
+The easiest way is to update your fork of `flux-helm-test` and
+change the `image` argument.
+
+Replace `YOURUSER` in `https://github.com/YOURUSER/flux-helm-test/edit/master/releases/mongodb_release.yaml`
+with your Github ID, open the URL in your browser, edit the file,
+change the `image:` line to the following:
+
+```yaml
+ image: bitnami/mongodb:3.7.9-r13
+```
+
+Commit the change to your `master` branch. It will now get
+automatically deployed to your cluster.
+
+You can check out the Flux logs with:
+
+```sh
+kubectl -n flux logs deployment/flux -f
+```
+
+The default sync frequency for Flux using the Helm chart is
+30 seconds. This can be tweaked easily. By observing the logs
```

Comment:
Goodness, is it? That's a bit of a departure from flux's actual default, which is a deliberately conservative five minutes.

Reply:
I had a discussion with @stefanprodan about it. Is there a way we could find a middle ground and reconcile the value across both worlds?

Reply:
I was thinking about dropping it down to `2m`, but I'd be reluctant to go further than that. Why? Because applying can plausibly on the order of a minute, and if you set the sync interval for less than it typically takes to apply, it'll do little else.

Polling git is another thing -- you might want to be more responsive to commits turning up there. 30s is more or less OK for that; but, better to have hooks set up and poll less frequently. Once hooks are available.

(The value in the Helm chart is used for both git and sync interval)

Reply:
I'm happy to document whatever is reality these days. Maybe @stefanprodan can comment on the above.